BinaryEncoding.md

```diff
@@ -234,6 +233,52 @@ This section must be preceded by a [Functions](#functions-section) section.
 | count | `varuint32` | count of entries to follow |
 | entries | `uint16*` | repeated indexes into the function table |
 
+### Function Names section
+
+This section may occur 0 or 1 times and does not change execution semantics. A
+validation error in this section is not reported and is treated as the section
+being absent. The expectation is that, when a binary WebAssembly module is
+viewed in a browser or other development environment, the names in this section
+will be used as the names of functions in the [text format](TextFormat.md).
+
+ID: `function_names`
+
+| Field | Type | Description |
+| ----- |  ----- | ----- |
+| names | `name*` | sequence of names |
+
+The number of names is determined by the number of declared functions. The
+sequence of names assigns a name to each function index. A name is:
+
+| Field | Type | Description |
+| ----- |  ----- | ----- |
+| size | varuint32 | string length, in bytes |
+| string | bytes | valid utf8 encoding |
```

Comment:
@jfbastien This was the point of the text above saying "A validation error in this section is not reported and is treated as the section being absent.": a WebAssembly implementation that doesn't care can simply skip over this section.  And if you *do* want to use the text, either it's definitely in utf8 (so you need the decoder) or it's not and we have the fractured-ecosystem-with-no-way-to-tell-what-the-encoding-is problem.